### ISSUE TITLE
[Bug] Authorization header specification using @openapi.parameter is not included Swagger UI request

### ISSUE THREAD

**Issue by dacasals** [sanic-org/sanic-ext#23]
**Describe the bug**
Authorization header specification  using @openapi.parameter is not included Swagger UI request.

The UI shows the Input field, but the CURL example does not include the header and the request made does not include the header either.

**Screenshots**
The screenshots of Swagger UI if this bug is related to it.
![image](https://user-images.githubusercontent.com/10888922/146428476-82d16eff-0eda-4aec-a4e6-10bce2063e85.png)

**To Reproduce**
``` python
from sanic.response import text
from sanic_ext import Extend
from sanic_ext.extensions.openapi import openapi

app = Sanic("MyHelloWorldApp")
Extend(app)

@app.get("/")
@openapi.parameter("Authorization", str, "header", required=True)
async def hello_world(request):
    return text("Hello, world.")


if __name__ == "__main__":
    app.run(
        host="0.0.0.0",
        port=8002,
        auto_reload=True,
    )
```
**Expected behavior**
I expect receive 
``` bash

curl -X 'GET' \
  'http://localhost:8002/ \
  -H 'accept: application/json' \
  -H 'Authorization: token123'
```
**Environment:**
 - OS:  MacOS
 - Browser  chrome
 - Version  sanic==21.9.3, sanic-ext==21.9.3

**Comment by ahopkins** [comment]
Odd. I am curious to know if it is that we are generating invalid OAS, or is an issue with the client. Do you know that this works in other Swagger installations?

**Comment by ahopkins** [comment]
I don't think this is related, but there is an open PR re: that decorator: https://github.com/sanic-org/sanic-ext/pull/21

**Comment by dacasals** [comment]
I found the problem and it is  specifically related with the Authorization header. 
The thing is in this case, as mentioned [here](https://swagger.io/docs/specification/describing-parameters/#header-parameters)
_"Note: Header parameters named Accept, Content-Type and Authorization are not allowed. To describe these headers, use the corresponding OpenAPI keywords"_


So for the Authorization header i think the sanic-ext  needs to implement the specification described [here.](https://swagger.io/docs/specification/authentication/)


So the ir no problem with the openapi.parameter  for others headers except  headers mentioned above.

In the case of Authorization, we need to generate something like this:

``` json
{"openapi": "3.0.0",
  "info": {
    "title": "API",
    "version": "1.0.0",
    "contact": {
      
    }
  },
  "components":
  { "securitySchemes":{
    "BearerAuth":{
      "type": "http",
      "scheme": "bearer"
    }
  }
    
  },

  "paths": {
    "/": {
      "get": {
        "operationId": "get_route_name",
        "summary": "Hello World",
        "security": [{"BearerAuth":[]}],
        "responses": {
          "default": {
            "description": "OK"
          }
        }
      }
    }
  },
  "tags": [
    
  ],
  "servers": [
    
  ]
}
```

**Comment by ahopkins** [comment]
Yup, that makes sense. That is a feature we need to add still: https://github.com/sanic-org/sanic-ext/blob/main/sanic_ext/extensions/openapi/openapi.py#L230

Hopefully we can squeeze this one into the upcoming release.

**Comment by dacasals** [comment]
Thanks,  in the mean time if someone  get stuck with this and found this thread,  you could do  a ugly hack like this:



``` python

# You must  add  security schemas to the singleton Builder of the specification..

app = Sanic("app", log_config=define_logging_config(LOGGING_CONFIG))
....
# Modify Singleton builder to add security schemes, ex: add a bearerAuth schema for bearer tokens :
SpecificationBuilder().add_component(
    "securitySchemes", "BearerAuth", {"type": "http", "scheme": "bearer"}
...
Extend(app)

```

``` python
from sanic_ext import openapi

# This hack is to override default behavior of sanic_ext openapi allow **secured** decorator. 
def secured(*args, **kwargs):
   #  raise NotImplementedError(
   #    "SecuritySchemas are not yet implemented in sanic-openapi 0.6.3, "
   #     "hopefully they should be ready for the next release."
   # )
    def inner(func):
        OperationStore()[func].secured(*args, **kwargs)
        return func

    return inner


openapi.secured = secured

```

``` python

# Finally annotate your secure route with:

@blueprint.route("/some_route", methods=["POST"])
@openapi.secured("BearerAuth")

```

Tested on sanic and sanic_ext  21.9.3

**Comment by ahopkins** [comment]
Working on this 

https://github.com/sanic-org/sanic-ext/pull/32

**Comment by ahopkins** [comment]
This is now released in v21.12.2 without @dacasals nice hack.

**Comment by ahopkins** [comment]
[Documentation](https://sanicframework.org/en/plugins/sanic-ext/openapi/security.html)
